src/rrd.c

```diff
@@ -919,34 +981,39 @@ unsigned long long rrdset_done(RRDSET *st)
 	now_ut  = st->last_collected_time.tv_sec * 1000000ULL + st->last_collected_time.tv_usec;
 	next_ut = (st->last_updated.tv_sec + st->update_every) * 1000000ULL;
 
-	if(unlikely(!first_entry && now_ut < next_ut)) {
-		if(unlikely(st->debug)) debug(D_RRD_STATS, "%s: THIS IS IN THE SAME INTERPOLATION POINT", st->name);
-	}
-
 	if(unlikely(st->debug)) {
-		debug(D_RRD_STATS, "%s: last ut = %0.3Lf (last updated time)", st->name, (long double)last_ut/1000000.0);
-		debug(D_RRD_STATS, "%s: now  ut = %0.3Lf (current update time)", st->name, (long double)now_ut/1000000.0);
-		debug(D_RRD_STATS, "%s: next ut = %0.3Lf (next interpolation point)", st->name, (long double)next_ut/1000000.0);
+		if(unlikely(!first_entry && now_ut < next_ut))
+			debug(D_RRD_STATS, "%s: THIS IS IN THE SAME INTERPOLATION POINT", st->name);
+
+		// FIX: Don't need 'long double' here.
+		debug(D_RRD_STATS, "%s: last ut = %0.3f (last updated time)", st->name, (double)last_ut/1000000.0);
+		debug(D_RRD_STATS, "%s: now  ut = %0.3f (current update time)", st->name, (double)now_ut/1000000.0);
+		debug(D_RRD_STATS, "%s: next ut = %0.3f (next interpolation point)", st->name, (double)next_ut/1000000.0);
 	}
 
 	if(unlikely(!st->counter_done)) {
 		store_this_entry = 0;
-		if(unlikely(st->debug)) debug(D_RRD_STATS, "%s: Will not store the next entry.", st->name);
+		if(unlikely(st->debug)) 
+			debug(D_RRD_STATS, "%s: Will not store the next entry.", st->name);
 	}
+
 	st->counter_done++;
 
 	// calculate totals and count the dimensions
-	int dimensions;
+	// FIX: local var dimensions not used!
+	//int dimensions;
 	st->collected_total = 0;
-	for( rd = st->dimensions, dimensions = 0 ; likely(rd) ; rd = rd->next, dimensions++ )
+	for( rd = st->dimensions /*, dimensions = 0 */ ; likely(rd) ; rd = rd->next /*, dimensions++ */ )
 		st->collected_total += rd->collected_value;
 
-	uint32_t storage_flags = SN_EXISTS;
+	int storage_flags = SN_EXISTS;
+
+	// NOTE: Second scanning of dimensions.
 
 	// process all dimensions to calculate their values
 	// based on the collected figures only
 	// at this stage we do not interpolate anything
-	for( rd = st->dimensions ; likely(rd) ; rd = rd->next ) {
+	for( rd = st->dimensions ; likely(rd) ; rd = rd->next ) { /* scans the dimensions again... */
```

Comment:
Yep... Just a little annotation to remind me to try to get rid of all 
these loops...

On 28-04-2016 14:16, Costa Tsaousis wrote:
>
> In src/rrd.c 
> <https://github.com/firehol/netdata/pull/323#discussion_r61466446>:
>
> >
> >  	// process all dimensions to calculate their values
> >  	// based on the collected figures only
> >  	// at this stage we do not interpolate anything
> > -	for( rd = st->dimensions ; likely(rd) ; rd = rd->next ) {
> > +	for( rd = st->dimensions ; likely(rd) ; rd = rd->next ) { /* scans the dimensions again... */
>
> it needs their total below.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/firehol/netdata/pull/323/files/460da7b0b8e59ebcb062b3b1627b452141c1a8ca..21b2e778bcce1d92816265f64175951556e53350#r61466446>
>